Improve error messages

zuhanit opened the pull request:
This changes `RuntimeError` messages to print looks like actual codes. See #54

examples:
1. Duplicate Section
<img width="199" alt="image" src="https://user-images.githubusercontent.com/36349353/163256228-8f809446-3a97-4ef0-8b0c-55e84e26eee9.png">
<img width="851" alt="image" src="https://user-images.githubusercontent.com/36349353/163256169-fc7565aa-a32b-4a30-8593-aa75b073b24f.png">

2. Duplicate Key

<img width="185" alt="image" src="https://user-images.githubusercontent.com/36349353/163255992-439f93cc-6d95-4872-8c3e-63f211471f42.png">
<img width="856" alt="image" src="https://user-images.githubusercontent.com/36349353/163255812-514166c6-6e41-412d-87fe-5a7d8d4f323b.png">

zuhanit committed:
Improve error messages

Change RuntimeError messages

armoha commented on readconfig.py:
```diff
@@ -27,6 +27,15 @@
 from collections import OrderedDict
 from typing import Dict
 
+class bcolors:
```
ANSI Escape Sequences are not enabled by default, and not supported before Windows 10 TH2 (v1511).
![image](https://user-images.githubusercontent.com/16814706/163305748-68312fd7-d3aa-4c96-ba0a-413ec03fc601.png)

armoha replied:
EE2 and EE3 examples

![image](https://user-images.githubusercontent.com/16814706/163307186-207f02e6-1764-4d9f-804c-64476e853256.png)
![image](https://user-images.githubusercontent.com/16814706/163307500-9c2477e3-d13d-4952-bab7-fa1fa3d32dee.png)

armoha commented on readconfig.py:
```diff
@@ -72,11 +84,15 @@ def parse_config(text):
                 value = keyvalue_match.group(3)
                 if key in currentSection:
+                    origin_lineno_length, duplicated_lineno_length = str(config_lineno[key]).__len__(), str(lineno).__len__()
+                    current_key_length = key.__len__()
```
We need to deal with non-ASCII characters too.
![image](https://user-images.githubusercontent.com/16814706/163306324-58998f24-ae45-4746-85fb-dc56ec004d64.png)

zuhanit replied:
I think it might be hard because fonts are vary in settings. Is there any idea to measure width of text?

```python
# It's hard to measure by using utf-8 bytes.
len('A'.encode('utf-8')) # 1
len('؄'.encode('utf-8')) # 2
len('하'.encode('utf-8')) # 3
len('𐰊'.encode('utf-8')) # 4
```

Instead, How about change form of error message to below?
<img width="368" alt="image" src="https://user-images.githubusercontent.com/36349353/163330102-91b83907-4d6d-4484-a787-22231d1d62ca.png">
<img width="326" alt="image" src="https://user-images.githubusercontent.com/36349353/163330132-a9968bf1-5df0-4553-a043-0f02e721f6f0.png">

armoha reviewed (changes requested):
Nice work! Few nitpicking.

armoha commented:
Using colorama as dependency would be simplest way to color terminal text: https://github.com/tartley/colorama
> ANSI escape character sequences have long been used to produce colored terminal text and cursor positioning on Unix and Macs. Colorama makes this work on Windows, too, by wrapping stdout, stripping ANSI sequences it finds (which would appear as gobbledygook in the output), and converting them into the appropriate win32 calls to modify the state of the terminal. On other platforms, Colorama does nothing.

zuhanit committed:
Change message form

Using colorama library instead of bcolors.
Dedent RuntimeError messages by textwrap.dedent().

armoha commented:
Sounds good to me, thanks for the PR!
